posthog/integrations/django.py

```diff
@@ -0,0 +1,116 @@
+from typing import TYPE_CHECKING, cast
+from posthog import scopes
+
+if TYPE_CHECKING:
+    from django.http import HttpRequest, HttpResponse  # noqa: F401
+    from typing import Callable, Dict, Any, Optional  # noqa: F401
+
+
+class PosthogContextMiddleware:
+    """Middleware to automatically track Django requests.
+
+    This middleware wraps all calls with a posthog scope. It attempts to extract the following from the request headers:
+    - Session ID as $session_id, (extracted from `X-POSTHOG-SESSION-ID`)
+    - Distinct ID as $distinct_id, (extracted from `X-POSTHOG-DISTINCT-ID`)
+    - Request URL as $current_url
+    - Request Method as request_method
+
+    The scope will also auto-capture exceptions and send them to PostHog, unless you disable it by setting
+    `POSTHOG_MW_CAPTURE_EXCEPTIONS` to `False` in your Django settings.
```

Comment:
How does this work alongside the exceptions Django integration? Is it possible that we could double capture exceptions? Maybe we should be folding them both into one

Reply:
Fair. Deffo think we should consolidate